nemo_aligner/algorithms/supervised.py

```diff
@@ -222,18 +240,10 @@ def save(self, extra_candidates=None, is_train_end=False):
         self.ckpt_callback.custom_save(monitor_candidates=monitor_candidates, is_train_end=is_train_end)
 
     def set_max_steps(self):
-        max_steps = self.cfg.get("max_steps", -1)
-
-        if max_steps == -1:
-            # the dataloader already knows how much longer
-            # because consumed samples is resumed
-            max_steps = self._train_dataloader_len
-        else:
-            # user specified the max step, figure out how much longer
-            # we need to run for
-            max_steps = max_steps - self.step
-
-        self.max_steps = min(max_steps, self._train_dataloader_len) + self.step
+        self.max_steps = (self._train_dataloader_len * (self.cfg.max_epochs - self.epoch)) + self.step
+
+        if (max_steps := self.cfg.get("max_steps", -1)) >= 0:
```

Comment:
can we not use the `walrus` operator to make it as compatible with other python versions as possible?(i know nemo doesn't support anything above python 3.10+) but I don't see much benefit in forcing a higher python version with this operator.

Reply:
The walrus operator was introduced in Python 3.8. But I don't understand why we would prevent ourselves from using Python features from the min version supported by NeMo.

Reply:
I don't have a hard opinion on this but I don't see the benefit of saving a few lines with the walrus operator if it means we can never support less than 3.8. NeMo containers need to be the most up to date everything, but we have more flexibility when it comes to our own dockerfile.

Reply:
Let's be sensible, nobody uses Python 3.7 anymore, it's >5y old, security support has ended already, and if we want to support it we also need to get rid of all f-strings in the code...

Reply:
ok good point, i don't have strong opinion on this so I'm ok with just 3.8+. Resolving